### PR TITLE
Forward `onload` and `error` events on itself,

### PR DESCRIPTION
**Pull request by tomalec**
Allows preventing FOUC in https://github.com/Starcounter/starcounter-include/issues/93

### CODE REVIEW

File: test/events.html [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            loadSpy = sinon.spy();
+            errorSpy = sinon.spy();
+            shadowHost = document.createElement('div');
+            shadowHost.attachShadow({mode:'open'});
+            shadowHost.shadowRoot.innerHTML = `<enlighted-link rel="stylesheet"></enlighted-link>`;
+            shadowHost.shadowRoot.innerHTML = `<enlighted-link rel="import"></enlighted-link>`;
+            enlightedLink = shadowHost.shadowRoot.querySelector('enlighted-link');
+        });
+        afterEach(function() {
+            shadowHost.remove();
```
Comment (alshakero) [inline test/events.html]: Isn't this discouraged as per your findings?

File: test/events.html [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            enlightedLink.removeEventListener('load', loadSpy);
+            enlightedLink.removeEventListener('error', errorSpy);
+        });
+        it(`\`load\` event when link is loaded`, function(done) {
```
Comment (alshakero) [inline test/events.html]: I suggest using a single quote to get rid of escaping.

File: test/events.html [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            enlightedLink.setAttribute('href', 'assets/doc-with-js.html');
+
+            setTimeout(()=>{
+                expect(loadSpy).to.be.calledOnce;
+                expect(loadSpy.args[0][0].type).to.equal('load');
+                done();
+            },100);
+            document.body.appendChild(shadowHost);
+        });
+        it(`only one \`load\` event when link is loaded for re-attached element`, function(done) {
```
Comment (alshakero) [inline test/events.html]: Same as above

File: test/events.html [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            enlightedLink.setAttribute('href', 'assets/doc-with-js.html?2');
+
+            setTimeout(()=>{
+                expect(loadSpy).to.be.calledOnce;
+                expect(loadSpy.args[0][0].type).to.equal('load');
+                done();
+            },100);
+            document.body.appendChild(shadowHost);
+        });
+        it(`\`error\` event when link is errored`, function(done) {
```
Comment (alshakero) [inline test/events.html]: Same as above

File: test/events.html [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            enlightedLink.setAttribute('href', 'non-existing.html');
+
+            setTimeout(()=>{
+                expect(errorSpy).to.be.calledOnce;
+                expect(errorSpy.args[0][0].type).to.equal('error');
+                done();
+            },100);
+            document.body.appendChild(shadowHost);
+        });
+        it(`only one \`error\` event when link is errored for re-attached element`, function(done) {
```
Comment (alshakero) [inline test/events.html]: Same as above

File: enlighted-link.html [hunk 6]
```diff
@@ -14,7 +14,17 @@
...
     connectedCallback(){
-        const link = this.link = this.link || document.createElement('link');
+        const forwardEvent = (oldEvent)=>{
+            this.dispatchEvent(new oldEvent.constructor(oldEvent.type, oldEvent));
+        }
+        let link;
+        if(this.link){
+            link = this.link;
+        } else {
+            link = this.link = this.link || document.createElement('link');
```
Comment (alshakero) [inline enlighted-link.html]: I think `this.link` in `this.link ||` will never evaluate to truthy. Because if it does, we won't make it to `else`.

File: README.md [hunk 7]
```diff
@@ -56,6 +56,17 @@ The element forwards [`link` element's content attributes](https://dev.w3.org/ht
 - sizes
 - title
 
+## Properties
+
+Property   | Options           | Default | Description
+---        | ---               | ---     | ---
+`link`     | *HTMLLinkElement* |         | Created `<link>` element, bare in mind that it may not be disconnected from `document.head`, if `<enlighted-link>` is disconnected, or `null` if the element was not connected at all.
```
Comment (alshakero) [inline README.md]: Can you rephrase this? It's a unclear.

**Review by alshakero — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (tomalec) [reply README.md]: What about:
> The reference to created `<link>` element. `null` if not yet created. Please note, that link element can be disconnected from `document` tree when `<enlighted-link>` is disconnected.

File: test/events.html (hunk 1 above)
Comment (tomalec) [reply test/events.html]: Which findings? Why should it be? I'm cleaning what I created in `beforeEach` to start clean.

Comment (alshakero) [reply test/events.html]: I mean `shadowHost.parentNode.removeChild(shadowHost)`.

File: README.md (hunk 7 above)
Comment (alshakero) [reply README.md]: Perfect!

**Review by alshakero — APPROVED** [review]
LGTM!

File: test/events.html (hunk 1 above)
Comment (tomalec) [reply test/events.html]: Ah. right! thx
